src/biogeophys/SnowSnicarMod.F90

```diff
@@ -1372,62 +1823,652 @@ subroutine SnowOptics_init( )
-     call ncd_io( 'ss_alb_dust02', ss_alb_dst2,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust02', asm_prm_dst2,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust02', ext_cff_mss_dst2, 'read', ncid, posNOTonfile=.true.)
-
-     ! dust species 3 Mie parameters
-     call ncd_io( 'ss_alb_dust03', ss_alb_dst3,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust03', asm_prm_dst3,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust03', ext_cff_mss_dst3, 'read', ncid, posNOTonfile=.true.)
-
-     ! dust species 4 Mie parameters
-     call ncd_io( 'ss_alb_dust04', ss_alb_dst4,           'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'asm_prm_dust04', asm_prm_dst4,         'read', ncid, posNOTonfile=.true.)
-     call ncd_io( 'ext_cff_mss_dust04', ext_cff_mss_dst4, 'read', ncid, posNOTonfile=.true.)
-
+     !--------------------- for 5-band data
+     select case (snicar_numrad_snw)
+     case (5)  ! 5-band case
+
+        select case (snicar_solarspec)
+        ! mid-latitude winter spectrum
+        case ('mid_latitude_winter')
+           ! flux weights/spectrum
+           call ncd_io( 'flx_wgt_dir5_mlw', flx_wgt_dir,     'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'flx_wgt_dif5_mlw', flx_wgt_dif,     'read', ncid, posNOTonfile=.true.)
+           ! BC species 1 Mie parameters, uncoated BC, same as bc2 before BC-snow internal mixing
+           call ncd_io( 'ss_alb_bcphob_dif_mlw', ss_alb_bc1,           'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'asm_prm_bcphob_dif_mlw', asm_prm_bc1,         'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'ext_cff_mss_bcphob_dif_mlw', ext_cff_mss_bc1, 'read', ncid, posNOTonfile=.true.)
+           ! BC species 2 Mie parameters, uncoated BC
+           call ncd_io( 'ss_alb_bcphob_dif_mlw', ss_alb_bc2,           'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'asm_prm_bcphob_dif_mlw', asm_prm_bc2,         'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'ext_cff_mss_bcphob_dif_mlw', ext_cff_mss_bc2, 'read', ncid, posNOTonfile=.true.)
+           ! OC species 1 Mie parameters, uncoated OC, same as oc2 before OC-snow internal mixing
+           call ncd_io( 'ss_alb_ocphob_dif_mlw', ss_alb_oc1,           'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'asm_prm_ocphob_dif_mlw', asm_prm_oc1,         'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'ext_cff_mss_ocphob_dif_mlw', ext_cff_mss_oc1, 'read', ncid, posNOTonfile=.true.)
+           ! OC species 2 Mie parameters, uncoated OC
+           call ncd_io( 'ss_alb_ocphob_dif_mlw', ss_alb_oc2,           'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'asm_prm_ocphob_dif_mlw', asm_prm_oc2,         'read', ncid, posNOTonfile=.true.)
+           call ncd_io( 'ext_cff_mss_ocphob_dif_mlw', ext_cff_mss_oc2, 'read', ncid, posNOTonfile=.true.)
```

Comment:
@cenlinhe in this and subsequent sections I noticed that you labeled bc1/oc1 and bc2/oc2 as `phob`. Is it intentional or should I correct?

Reply:
I did this intentionally. bc1/oc1 is assumed to have the same optics as bc2/oc2, because sulfate coating on the bc1/oc1 surface is assumed to be dissolved into the hydrometeo (i.e, snow grain here) during deposition process. This is different from the previous assumption, where bc1/oc1 is coated by undissolved sulfate.

Reply:
If the bc-snow internal mixing option is turned on, then bc1/oc1 (internally mixed within snow grain) will be treated differently than bc2/oc2 (mixed externally or outside snow grain). This treatment is done in the snicar code, not the input data here.

Reply:
> This is different from the previous assumption, where bc1/oc1 is coated by undissolved sulfate.

To be 100% clear, by the previous assumption, are you referring to the earlier version of the model or earlier lines of code in this version?

Reply:
I am referring to the earlier version of the model.

Reply:
Thank you for your patience @cenlinhe. I wish to clarify one more thing...

I now understand that:
- later in the code, in snicar, hydrophillic particles (bc1/oc1) are internally mixed and hydrophobic particles (bc2/oc2) are not internally mixed
- in this section of code, hydrophobic particles behave like hydrophillic particles (sulfate coating dissolved rather than undissolved)

So my question is this: Should _bc1/oc1 and _bc2/oc2 variables be assigned the bcphil parameters rather than the bcphob parameters?

Reply:
No. In fact, in this section of code, hydrophillic particles behave like hydrophobic particles. After sulfate coating dissolved, what remains the the core BC particle which is hydrophobic. The optical property for bcphil in the snicar input database assuming bc has the sulfate coating.